upload/fromurl.go

```diff
@@ -234,20 +234,35 @@ func (d *fromURLData) wait() {
 				data,
 			)
 
-			if data.Status == uploadStatusError {
+			switch data.Status {
+			case uploadStatusSuccess:
+				d.done <- *data.FileInfo
+				return
```

Comment:
ideally I'd like to have a guard here if `data.FileInfo` is still `nil` for some reason.
maybe crash, but still leave some info to debug.

Reply:
added some more ugly checks :)

Reply:
let's roll this out finally